### PR TITLE
[extractor/cda] Fix age restriction

### LINKED ISSUE #5980: [CDA] Unable to download age restricted videos
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a broken site
- [x] I've verified that I'm running yt-dlp version **2023.01.06** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Should be worldwide, Poland

### Provide a description that is worded well enough to be understood

Age restricted videos causes the error provided in output section upon the download attempt.

Example video for reproducing the issue: https://www.cda.pl/video/1977375ab

This seems to be caused by the website UI change. I do not frequent it too much so I'm not 100% sure about this but I believe that back in the day, it used to ask the user to fill a birthday validation form upon trying to access the age restricted video. However right now this is replaced with a different and much simpler form with only a single confirmation button:
```
<div class="pd-player">
  <div class="clear"></div>
  <div class="boxAdult">
      <div class="ageConfirm">
          <div class="contentAdult">
              <div style="padding-top: 18px;">
                  <h2 style="text-align:center">Materiał przeznaczony dla osób pełnoletnich</h2>
                  <form id="upload_form" enctype="multipart/form-data" method="post">
                      <div class="form-row form-table" style="margin: 12px auto 0;text-align:center"></div>
                      <p style="text-align:center;padding-top: 15px;">
                          <button name="age_confirm" class="btn btn-my" style="margin-top:10px;margin-bottom:30px">Mam ukończone 18 lat lub zgodę opiekuna prawnego.</button>
                      </p>
                  </form>
                  <p style="padding:20px 20px 20px">Niniejsza treść została uznana przez Użytkowników lub Administratora za treść nieodpowiednią dla osób małoletnich. Materiał może zawierać treści wrażliwe, w szczególności pornografię lub w sposób nieuzasadniony eksponując przemoc. Aby obejrzeć materiał Użytkownik musi mieć ukończone 18 lat lub posiadać zgodę przedstawiciela ustawowego na zapoznanie się z niniejszą treścią. Usługodawca zastrzega, że zgodnie z &nbsp;treścią Regulaminu, Użytkownikiem Serwisu może być wyłącznie osoba pełnoletnia lub małoletnia działającą za zgodą przedstawiciela ustawowego. Zapoznanie się z treścią z naruszeniem tych zasad następuje na wyłączną odpowiedzialność Użytkownika.
                  </p>
              </div>
          </div>
      </div>
  </div>
  ...
</div>
```

As the old form is gone completely, the [condition responsible for bypassing the age restrictions](https://github.com/yt-dlp/yt-dlp/blob/master/yt_dlp/extractor/cda.py#L212-L213) is never fulfilled thus causing the failure down the way. I have little Python experience and I'm not good with regexps but in order to confirm my suspition, I've tried changing it to look for the new button which seems to make it work again:
```
if self._html_search_regex(r'(<button[^>]+name="[^"]*age_confirm[^"]*")',
                                   webpage, 'birthday validate form', default=None):
```

Quite bizarrely after the condition is fixed, the current  `_download_age_confirm_page` function seems to still do its job just fine. Not sure if that's an oversight on site's side or they allow the old method for some compatibility reason or whatnot. What currently happens in-browser however, after user clicks the confirmation button the resource is re-requested with much smaller payload (source for this is Chrome Devtools' "Payload" tab):
```
------WebKitFormBoundaryMBSyCKEk13omZl75
Content-Disposition: form-data; name="age_confirm"


------WebKitFormBoundaryMBSyCKEk13omZl75--
```
I weren't able to spot any redirect containing the `/a/validatebirth` URL and most certainly the birthdate information is nowhere to be present within the request. This made me believe that it would probably also be nice to update the function to something more reflective of what's currently happening and so I've tried replacing it with simply:
```
def _download_age_confirm_page(self, url, video_id, *args, **kwargs):
        form_data = {'age_confirm': ''}
        data, content_type = multipart_encode(form_data)
        return self._download_webpage(
            url, video_id, *args,
            data=data, headers={
                'Referer': url,
                'Content-Type': content_type,
            }, **kwargs)
```

Which seems to still work fine. Finally, I do not own an account for the site and can't say whether logged-in users are able to bypass the issue. Since I've tried the `--cookies-from-browser` flag, can confidently say it doesn't help for unauthorized ones. 

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['https://www.cda.pl/video/1977375ab', '-vU']
[debug] User config: []
[debug] System config: []
[debug] Encodings: locale cp1250, fs utf-8, pref cp1250, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2023.01.02 [d83b0ad] (win32_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.19044-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg 2021-11-22-git-203b0e3561-full_build-www.gyan.dev (setts), ffprobe 2021-11-22-git-203b0e3561-full_build-www.gyan.dev
[debug] Optional libraries: Cryptodome-3.16.0, brotli-1.0.9, certifi-2022.12.07, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Loaded 1754 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2023.01.02, Current version: 2023.01.02
yt-dlp is up to date (2023.01.02)
[CDA] Extracting URL: https://www.cda.pl/video/1977375ab
[CDA] 1977375ab: Downloading webpage
WARNING: [CDA] unable to extract default player_json; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
ERROR: [CDA] 1977375ab: No video formats found!; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
Traceback (most recent call last):
  File "yt_dlp\YoutubeDL.py", line 1497, in wrapper
  File "yt_dlp\YoutubeDL.py", line 1594, in __extract_info
  File "yt_dlp\YoutubeDL.py", line 1653, in process_ie_result
  File "yt_dlp\YoutubeDL.py", line 2601, in process_video_result
  File "yt_dlp\YoutubeDL.py", line 1038, in raise_no_formats
yt_dlp.utils.ExtractorError: [CDA] 1977375ab: No video formats found!; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
```

### PR DESCRIPTION
**Pull request by Szpachlarz**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

This pull request updates age confirmation to work with the new form. The code is as it was proposed by @emqi, the original poster of #5980.

Fixes #5980

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [ ] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [x] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

**Comment by pukkandan** [comment]
Have you run the extractor download tests? Pls update test defenitions as needed

**Comment by Szpachlarz** [comment]
I've fixed the tests and removed some dead code.

File: yt_dlp/extractor/cda.py [hunk 1]
```diff
@@ -63,38 +62,39 @@ class CDAIE(InfoExtractor):
...
-            'view_count': int,
             'average_rating': float,
             'duration': 137,
             'age_limit': 0,
+            'upload_date': '20160220',
+            'timestamp': 1455968218,
         }
     }, {
         # Age-restricted
-        'url': 'http://www.cda.pl/video/1273454c4',
```
Comment (selfisekai) [inline yt_dlp/extractor/cda.py:73]: both are piracy, replace with https://www.cda.pl/video/8753244c4/vfilm

**Review by selfisekai — CHANGES_REQUESTED** [review]

**Comment by dirkf** [comment]
A URL without `/vfilm` redirects to the `/vfilm` URL but the redirected URL doesn't pass the age confirmation because the extractor is still sending the original URL.
```diff
     def _web_extract(self, video_id, url):
         self._set_cookie('cda.pl', 'cda.player', 'html5')
         player_url = '{0}/video/{1}/vfilm'.format(self._BASE_URL, video_id)
-        webpage = self._download_webpage(player_url, video_id)
+        webpage, urlh = self._download_webpage_handle(player_url, video_id)
+        url = urlh.geturl()

```

**Comment by Lupenus** [comment]
So is this thread dead or something?

**Pull request closed without merging**